### PR TITLE
Create smoke test for Snowflake notebooks.

### PR DESCRIPTION
**Pull request by sfc-gh-dkurokawa**
# Description
Create smoke test for Snowflake notebooks.

This test is currently failing since go/mlplatformtest doesn't have Snowflake notebooks working for some reason.

## Other details good to know for developers
Future work is to test a lot more notebooks, but this is the framework for testing them. I also wanted to get this out first (aside from just not having huge PRs) since this is already broken behavior and want to have this test to more easily test when it's fixed.

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
  not work as expected)
- [x] New Tests
- [ ] This change includes re-generated golden test results
- [ ] This change requires a documentation update

----

> [!IMPORTANT]
> Add smoke test for Snowflake notebooks and update Snowflake test utilities for schema management.
> 
>   - **New Tests**:
>     - Add `test_snowflake_notebooks.py` to test Snowflake notebooks by uploading and executing a simple notebook.
>     - Introduce `environment.yml` for Snowflake dependencies.
>   - **Utilities**:
>     - Modify `SnowflakeTestCase` in `snowflake_test_case.py` to use a `set` for `_snowflake_schemas_to_delete` and add `create_and_use_schema()` method.
>     - Update `run_query()` in `snowflake_test_case.py` to accept query bindings.
>   - **Bug Fixes**:
>     - Fix `test_snowflake_connection.py` to use `set` for `_snowflake_schemas_to_delete`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=truera%2Ftrulens&utm_source=github&utm_medium=referral)<sup> for 8ce6ec82333964eb62cbf990c4acd51a7b9eedbf. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: tests/e2e/data/environment.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+name: app_environment
+channels:
+  - snowflake
+dependencies:
+  - trulens-connectors-snowflake=*
```
Comment (sfc-gh-chu) [inline tests/e2e/data/environment.yml:5]: do we need to create an environment.yml for each test case/suite?

File: tests/e2e/test_snowflake_notebooks.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            self.run_query(
+                f"""
+                CREATE NOTEBOOK {name}
+                FROM '@{_STAGE_NAME}'
+                MAIN_FILE = '{name}.ipynb'
+                QUERY_WAREHOUSE = {self._snowflake_connection_parameters["warehouse"]}
+                """
+            )
+            self.run_query(f"ALTER NOTEBOOK {name} ADD LIVE VERSION FROM LAST")
+            self.run_query(f"EXECUTE NOTEBOOK {name}()")
```
Comment (sfc-gh-chu) [inline tests/e2e/test_snowflake_notebooks.py:41]: how does this handle notebook errors?

File: tests/e2e/data/environment.yml (hunk 1 above)
Comment (sfc-gh-dkurokawa) [reply tests/e2e/data/environment.yml:5]: No we can reuse the same one if we want. I'm currently just running the single simple notebook and just testing all the imports and printing out some obvious statement.

File: tests/e2e/test_snowflake_notebooks.py (hunk 2 above)
Comment (sfc-gh-dkurokawa) [reply tests/e2e/test_snowflake_notebooks.py:41]: The error is propagated as an `snowflake.snowpark.exceptions.SnowparkSQLException` but the error message will say what was wrong. Here's an example of me running it where I didn't install the right packages:
```
E       snowflake.snowpark.exceptions.SnowparkSQLException: (1304): 01b816c0-0004-1bcf-0038-7187e48d721a: 092848 (42601): User app execution error Python Interpreter Error:
E       Exception: ModuleNotFoundError: Line 1: Module Not Found: trulens.connectors. To import packages from Anaconda, install them first using the package selector at the top of the page.
E       .

../../../../miniconda3/envs/trulens_3_11/lib/python3.11/site-packages/snowflake/connector/errors.py:215: SnowparkSQLException
```

File: tests/e2e/test_snowflake_notebooks.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        finally:
+            self.run_query(f"DROP STAGE IF EXISTS {_STAGE_NAME}")
```
Comment (sfc-gh-chu) [inline tests/e2e/test_snowflake_notebooks.py:43]: should we drop the schema here too?

Comment (sfc-gh-dkurokawa) [reply tests/e2e/test_snowflake_notebooks.py:43]: that happens automatically when you run `create_and_use_schema`.

Comment (sfc-gh-chu) [reply tests/e2e/test_snowflake_notebooks.py:43]: Oh my bad missed that

**Pull request merged**